R/get.samples.R

```diff
@@ -46,19 +57,27 @@ get.samples <- function(catch, lf, al, tresh.al = 2, tresh.lf = 2, period.unit =
     
     period.unit <- match.arg(period.unit)
     
-    # 2) Minor calculations and variable creation
-    options(dplyr.summarise.inform = FALSE) # https://www.tidyverse.org/blog/2020/05/dplyr-1-0-0-last-minute-additions/
-    lf <- lf %>% 
-          group_by_at(lfcol[1:5]) %>% 
-          mutate(prop = n/sum(n)) %>% # proportion of each length in each sample
-          as.data.frame()
+    if(prob.al<0 | prob.al>1) warning("prob.al should be restricted to 0-1")
     
+    # 2) Minor calculations and variable creation
+    id <- as.factor(apply(lf[,lfcol[1:5]],1,paste,collapse=""))
+    lf$prop <- ave(lf$n, id, FUN = function(z) z/sum(z))
```

Comment:
you just wanted it to fit one 1 line?

Reply:
yes :D

it's shorter, faster and uses baseR (important for package stability)